akka-stream/src/main/scala/akka/stream/impl/io/TcpListenStreamActor.scala

```diff
@@ -86,6 +86,11 @@ private[akka] class TcpListenStreamActor(localAddressPromise: Promise[InetSocket
         unbindPromise.success(() ⇒ { target ! Unbind; unboundPromise.future })
         primaryOutputs.getExposedPublisher.subscribe(flowSubscriber.asInstanceOf[Subscriber[Any]])
         subreceive.become(running)
+      case CommandFailed(_, Some(ex)) ⇒
+        localAddressPromise.failure(ex)
+        unbindPromise.failure(ex)
+        flowSubscriber.onError(ex)
+        fail(ex)
```

Comment:
Here it is particularly important to use the other implementation since it contains a fix for Reactive Streams compliance (the `tryOnError` part).